logstash-core/lib/logstash/util/wrapped_acked_queue.rb

```diff
@@ -59,7 +59,7 @@ def write_client
     end
 
     def read_client()
-      ReadClient.new(self)
+      LogStash::AckedReadClient.create(self)
```

Comment:
One thing I was not sure about here is whether all clients of `AckedReadClient` subsequently call `set_batch_dimensions` with explicit values. If not, this `create` method does not initialize the batch size and timeout while the `initialize` method of the Ruby implementation provides defaults for the batch size and timeout.

Reply:
@danhermann right ... thanks for spotting that one :) Fixing.

Reply:
@danhermann nevermind, this is correct. Sorry for the noise, see https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/java_pipeline.rb#L368 => the existing constructor was redundant => we're good here I think :)
... that is not to say that it isn't horrible that we're somehow setting the `@queue` in the constructor and the timeout values via some setter, but that's the next step :D